### DISCUSSION TITLE
[NixOS]: WaybarStyle and Layout script cant show file name

### DISCUSSION THREAD

**Discussion by solitudealma** [JaKooLit/Hyprland-Dots#230] [Q&A]
### Description

WaybarStyle and Layout script cant show file name. I think the problem is appear from read file name. I don’t know if you using the read command incorrectly, or there is something wrong with my use the config. Another script, WallPaperSelect, can be used normally because it uses ls to list the file names and stores them in an array. So I want you to see where this problem occurs.

### Steps to Reproduce

1. run the script
2. chose style

### Expected Behavior

to show all the style of waybar

### Actual Behavior

noting to show

### Screenshots

![image](https://github.com/JaKooLit/Hyprland-Dots/assets/57446969/94a27178-778d-4f9d-a763-58953d859d7c)

### Environment

- Linux Distro: [nixos]
- Linux Version, output of `uname -a`:
  `Linux laptop 6.8.2-zen2 #1-NixOS ZEN SMP PREEMPT_DYNAMIC Tue Jan  1 00:00:00 UTC 1980 x86_64 Toybox`
- Hyprland Version, output of `hyprctl version`: `Hyprland, built from branch  at commit 1c460e98f870676b15871fe4e5bfeb1a32a3d6d8`
- Installation Environment: On The Main Hardware

**Comment by JaKooLit** [comment]
it is reading the list on ~/.config/waybar/configs and ~/.config/waybar/styles

if your configs are in .nix, it wont read

**Comment by solitudealma** [comment]
> it is reading the list on ~/.config/waybar/configs and ~/.config/waybar/styles
> 
> if your configs are in .nix, it wont read

I konw your mean. I just use xdg.configFile to link. if i input the name of file, it will be selected. but the list of file name is empty. i cant see anything.

**Comment by JaKooLit** [comment]
![Screenshot_03-May_09-24-28_25962](https://github.com/JaKooLit/Hyprland-Dots/assets/85185940/f5213728-415d-430e-86bf-6d94840ddb17)
![Screenshot_03-May_09-24-36_20977](https://github.com/JaKooLit/Hyprland-Dots/assets/85185940/0740fd31-ab59-46f7-a8c5-2b8a493e3514)


I dont have issue on my nixos.. but then, I dont use home manager... 

If you have find some error to make it work for you, Im all open ears

**Comment by solitudealma** [comment]
nice, It will be easier if you also use nixos. As you think, I used hm to manage the configuration. But I just use xdg.configFile to link the configuration to the .config directory. This should be exactly the same for you. eg:
```shell
xdg.configFile."waybar" = {
   source = ./xxx;
   recursion = true;
}
```
So I'm not quite sure where the problem is, I deleted the soft linked files because it would affect the nixos build, but the problem is definitely not here. I try in
```shell
while IFS= read -r file; do
         if [ -f "$waybar_styles/$file" ]; then
             options+=("$(basename "$file" .css)")
         fi
     done < <(find "$waybar_styles" -maxdepth 1 -type f -name '*.css' -exec basename {} \; | sort)
```
The options are output in this code, but the desired result is not obtained. If you need, I can upload my nixos configuration so that you can reproduce this problem.

**Comment by JaKooLit** [comment]
I moved this in discussion

I dont have this

`xdg.configFile."waybar" = {
   source = ./xxx;
   recursion = true;
}
`

in case youre wondering, here is my nixos configs
https://github.com/JaKooLit/NixOS-configs

i just use flakes so I can easily edit my .nix but

**Reply by solitudealma** [reply]
ok, thx very much
